packages/ember-runtime/lib/computed/computed_macros.js

```diff
@@ -20,7 +20,10 @@ function expandPropertiesToArray(properties) {
   }
 
   for (let i = 0; i < properties.length; i++) {
-    expandProperties(properties[i], extractProperty);
+    let property = properties[i];
+    assert('Dependent keys passed to Ember.computed.and() or Ember.computed.or() can\'t have spaces.', property.indexOf(' ') < 0);
```

Comment:
Adding a second argument `label` to `generateComputedWithPredicate` and `expandPropertiesToArray` should work out well. It seemed a little awkward to me at first, but it leads to better messages for the user. If they've used both `and` and `or` in their app, this will help them narrow it down.